vembrane/backend/base.py

```diff
@@ -0,0 +1,272 @@
+from abc import abstractmethod, abstractproperty
+from enum import Enum
+from typing import List, Optional, Tuple
+
+from ..ann_types import NA
+from ..errors import UnknownSample
+
+
+class Backend(Enum):
+    pysam = (0,)
+    cyvcf2 = 1
+
+    def __str__(self):
+        return self.name.lower()
+
+    def __repr__(self):
+        return str(self)
+
+    @staticmethod
+    def from_string(s):
+        try:
+            return Backend[s]
+        except KeyError:
+            return s

```

Comment:
No this is a hack to make argparse properly let you choose from enums, that are adressed by string. I don't really know whats going on here, but if you raise an error instead of s, argparse doesn't print the correct information anymore in case the user doesn't choose correctly.

Reply:
Ah, I see, I didn't know that!